### ISSUE TITLE
Suggestion to display servers between 24 and 48 players

### ISSUE THREAD

**Issue by thereis** [SmartlyDressedGames/Unturned-3.x-Community#192]
One point we need to consider, it is to enable players to see servers between 24 and 48 players.

Unturned supports servers up to 48 players, but limiting them into a half of this size, it blocks them to join in a more competitive server and more enjoyable. They need to be advised about possibly performance problems before join in a "large server type".

Of course, this option, for we, servers owners, it will increase the players in our servers, more profit and earnings for us/unturned and it will decrease the amount of duplicated servers running in the same map, same plugins, same configuration, shared database, shared resources, etc...

Adding a option to display large servers in the Servers List, it will help them to navigate between the old/new feature.

You @SDGNelson need to review this and discuss with others servers owners/players and let this feature to be on Alpha for a month or a week for a test purpose. Please, consider this purposal.

**Comment by KlaasM** [comment]
Hello @thereis ,

As Nelson mentioned earlier "Servers with >24 players are pretty janky .."
It's not a good idea to show those janky servers by default; a filter would absolutely be required.

**Comment by thereis** [comment]
Yes, by default is not recommended, but we are in 2018 and this must be shown!

I don't want to use a plugin that everytime changes my current players to be "acceptable" on the Servers Lists.

Made it like Rust, Counter-Strike, 7 days to die... etc, they don't have any issues to list the large servers on the list, but this rule done by @SDGNelson, I think need to be reviewed...

**Comment by Jim-Lahey** [comment]
@thereis @KlaasM @SDGNelson 

"It's not a good idea to show those janky servers by default" This seems like a cop-out to me, why not fix the problem of the servers not displaying past 24 players **in the first place?**

Games with **less momentum, revenue, and support** are capable of allowing more than 24 players, on the same engine, same hardware, Why not Unturned as well? Is it **_really_** that difficult to allow support for more players than 24 at a time? Probably not, although I could be misinformed.

 That being said, the fact that we have to rely on _private, third-party, plugins_ to achieve a higher player count and show in the server list, is a sure sign of pure and utter laziness on Nelson's part, or just merely incompetence, ineptitude, or inability to care about this issue or Unturned itself even, does **anybody** even care?

 **Why do we need to rely on third-party developers for CRUCIAL and IMPORTANT fixes to the game?**
(Players beyond max 24)

 For lack of a better phrase, it seems to me like **Nelson just don't give a f*ck**. This is also evidenced by the fact the game breaks EVERY. SINGLE. FRIDAY.

This is what  millions of dollars in skin box sales looks like everybody, a janky broken ass game that hari-karis itself every Friday. Lit.

Please dude, test your patches. I can't speak for all of us, but I for sure am tired of dealing with broken shit every week: 

(Flying Mushrooms, Unturned: Ban Edition, Deleting Bundles folder in the deployment like a novice, adding Community folder to the global deploy, just to name a few examples)

 I could go on for ages, about issues caused by Nelson himself, more often than not caused by some "new cool feature", that inevitably breaks the game. I'll try to list what I can remember from my experiences:

-- max_packets, kicking everybody from servers by default, 
--The 1 month period where Unturned rocket installation forced to gold servers automatically, and no changes to commands.dat affected it. Evident collusion between Sven and Nelson.
-- Constant queue position #1 errors caused by lighting.dat
-- January, 2017 where Lighting.dat was causing mass airdrops on non-rocket servers.
-- Flying Mushroom incident
-- Added "Community" folder to Modules in global deployment, causing "This server is using a custom module asset you don't have."
-- Dumptruck vehicle asset incident
-- Mass battleye ban day
-- STILL NOT FIXING PEOPLE PUTTING BEDS OUTSIDE OF THE MAP. (Glitched bases)

I got a bit off base here, but this line of thinking stems from two full years, of hosting Unturned servers as my full time job. At one point I would dream about Unturned, it was the sole focus of my life. Please don't think I'm here just to be negative, I am completely at my wits' end in terms of tolerating this broken-ass shit any longer.

Let's put the issue of having only 24 players at a time, in perspective here:

-Maps being made, such as Greece, are **massive**, and don't take into account the fact every server will be limited to 24 players. It's almost as if there is not a single bit of oversight at all, in terms of how many players there will be on the map versus the distance they need to travel. These maps would be good for 50-100 players. Why is raising the player cap a non-issue? **IT MAKES NO SENSE!**

To put it plainly: Unturned is what it is because of **us** private people, who have a passion for the game, or money, or whatever, and choose to develop within Unturned in some way, by making a map, model, texture, plugin, or what have you. You owe it to us, to receive our feedback and respond to it in a tactful way. Sadly we are often skipped over quickly, with responses like "Servers with >24 players are pretty janky" response. **Wow, Klass. Thanks for the insight!** I know you're not Nelson himself but let's be real: What kind of advice is this? Servers above 24 players are janky. We're all aware of that.... but what about any plans to fix this issue? It seems like a pretty critical issue to me, but maybe I'm just stupid.

  Not to be rude, but wouldn't it be prudent to take the time to speak with the people who develop your game for you on a personal level? You would get valuable feedback that would help you to avoid these problems in the future. I seriously questions if millions of dollars in skins sales overwhelm Nelson to the point of anxiety? Either way, this is what it looks like when you pay over $1,000,000 to SDG via buying loot boxes: A frustrating, laggy, and Janky game that hari-karis itself every Friday.

  The workshop surpasses the actual game's content by far. I get that there are a lot of crowdsourced things in games, but take a cue from the workshop creators and make some interesting guns and vehicles. These dudes in their private homes are able to crank out high-quality and awesome mods. Wtf is actually happening behind the scenes if a bunch of kids making models, are better than the game itself? The max player count issue is a perfect example of this.

Back to the issue of the player count..

I would like to point out, the maps that are being "officially" released, aren't even made with the player count, or resources used in mind. This is a pain for every server host everywhere, because we can run 10x Rust servers for the same resource usage as Unturned. Why is this still a problem? Do you not realize that we keep Unturned alive by hosting servers in it? Surely, not being accommodating to our suggestions will spell death for Unturned in the long-run. Something stinks at SDG and we all smell it.

Is it obvious to anybody else the maps are not made with the end user, or even the host in mind? Or even the game itself? Max 24 player count isn't adaptable at all to any sizable map. We're talking 40 minutes in between running and finding somebody to shoot at. Wouldn't the game be more fun for everybody if it was conducive to more PvP action? More players, would definitely solve this issue.

The level of incompetence at SDG is also evidenced by these recent patches that add files that don't need to be there, delete folders that NEEDED to be there, etc. It's as if @SDGNelson **doesn't even test his patches before releasing them.** Wow. How can this be acceptable behavior from a gaming company that achieves millions of dollars in yearly revenue? This is a sign of laziness, or just pure incompetence. (Unturned: Ban Edition is a perfect example.)

For example, we could see how these max player count, lobby plugins are working, and use the same functionality (Displays 23/24, but actually has many more players.) Why has this not been thought of? Duh.

If I could give some advice: _Leave Unturned Alone_. Please don't give us _any new anything_, just take your money, and go, dude. We are perfectly capable of running the game without you. You have NO IDEA the far reaching impact of 4.0. You wasted all of our time and hard work because of your ineptitude. I would much prefer a game that is stable, than one that has new stuff and breaks every patch.

I can't get excited about 4.0 along with nearly every content creator, because after all this time we've spent making awesome things for your game, you're going to throw it all away to re-make the game on Unreal. Wtf?

It's a little late to decide this, do you even realize the butterfly-effect that this will have on your community? Have you stopped to take the time to speak with workshop creators and what their opinions are? If you did, I think you would be shocked. You effectively killed 90% of the developers making things for Unturned, by making it a frustrating environment to develop in.

I implore you, reach out to your colleagues and get their honest opinion on the current state of Unturned.

You're throwing away thousands if not hundreds of thousands of hours of work from people all around the planet, who may have just wanted Unturned to be the best game in the world, who put lots of love and time into their work. How could you not consider this?

In summary, posting any issues here is fruitless, because issues in Unturned are solved on a "if I feel like fixing it" basis.

On a side note,, can we address the problem of respawning in your bed to glitch out of the map? It has been a problem for far too long. Glitched Bases. They are on every single server, and can be done with completely vanilla mechanics and no plugins. My point is proven by this fact, I feel like. Why the heck haven't we fixed Glitched Bases yet?

As a player, I love Unturned, it's a simple and fun shooting game even a novice player can get into and enjoy. That being said, anybody who has spent some time to develop in it, knows it's a heaping pile of garbage. The past 2 years of making Unturned my full-time job has been equal parts stress and enjoyment, I'm really close to just being done with this janky-ass "game" filled with sponsorship, and collusion.

TL;DR Test your patches, fix glitched bases, raise player count, and most of all, please just listen to what we have to say from here on out if you want your game to survive.

**Comment by Trojaner25** [comment]
> Why do we need to rely on third-party developers for CRUCIAL and IMPORTANT fixes to the game?
These plugins do not fix anything. They only bypass limitations set by Nelson. Since this is an intended behaviour you can not say this is a bug.

> STILL NOT FIXING PEOPLE PUTTING BEDS OUTSIDE OF THE MAP.
This is something that is hard to fix. Only ugly workarounds exist which Nelson does not want to add.

>I got a bit off base here, but this line of thinking stems from two full years, of hosting Unturned servers as my full time job.
You shouldn't make hosting of Unturned servers your full time job, that's your problem.

>You owe it to us, to receive our feedback and respond to it in a tactful way. 
He gave you this game for free, he gave you 2 years of free enjoyment and you still think he owes you something? He has set up this GitHub issue tracker to take feedback and he responds to every single of them (just verrryy slowly). 

>Wtf is actually happening behind the scenes if a bunch of kids making models, are better than the game itself? The max player count issue is a perfect example of this.
Like said, player count issue is intended behaviour, the plugins do not fix anything here. They just raise it above the recommended limits.

>This is a pain for every server host everywhere, because we can run 10x Rust servers for the same resource usage as Unturned.
Unturned had very poor performance in the past and a lot of memory consumption. This is related to bugs and memory leaks in the Unity version used. However the latest updates fixed this issue. A friend of mine can host 8 full 24 slot servers on 2 core + 4 GB RAM.

>The level of incompetence at SDG is also evidenced by these recent patches that add files that don't need to be there, delete folders that NEEDED to be there, etc. It's as if @SDGNelson doesn't even test his patches before releasing them. 
These files were meant to be there since it was a new feature. However they were not registered to Battleye yet so it caused the kickings. As a hotfix Nelson removed it, but he will readd it again.

>For example, we could see how these max player count, lobby plugins are working, and use the same functionality (Displays 23/24, but actually has many more players.) Why has this not been thought of? Duh.
Those "lobby" plugins mess with player count which is also abused by other plugins to show 234/24 servers which actually have no players online.

**Comment by Jim-Lahey** [comment]
Well Trojaner, the fact you would defend all of this, knowing damn well that you refuse to make a plugin for this game again, just shows you are probably on the payroll. You can argue points here in text, but please provide evidence against anything I said not being true. Thanks.

Sad that we are reduced to collusion, it's obvious by Trojaner's comment he also works for Nelson. Probably signed an NDA and is on the payroll.

Can we deal with the real issues of this game, instead of deflection and FUD tactics?

**Comment by Trojaner25** [comment]
> Well Trojaner, the fact you would defend all of this, knowing damn well that you refuse to make a plugin for this game again, just shows you are probably on the payroll. 
WTF??
I will not agree on someone who is spreading false or wrong information.

> Sad that we are reduced to collusion, it's obvious by Trojaner's comment he also works for Nelson.
I wish I could work for Nelson and gain 150.000$ at the same time.

Anyway, just because someone defends someone else it doesnt mean they work for them or they are on a payroll..
This is getting ridiculous..

**Comment by Jim-Lahey** [comment]
> These plugins do not fix anything. They only bypass limitations set by Nelson. Since this is an intended behaviour you can not say this is a bug. People who want to bypass the recommended limits need to use third-party plugins.

If this is an "Intended Limitation", I'll be straight with you. IT'S STUPID. If the intended amount of players is 24, it just further proves my point, about the maps being designed without the player count in mind.

> This is something that is hard to fix. Only ugly workarounds exist which Nelson does not want to add.

Glitched bases are hard to fix? That is why a third-party developer made a plugin to stop this happening, within a couple weeks, right? It works flawlessly for me without issue. Why can't Nelson fix it? Lol.

> You shouldn't make hosting of Unturned servers your full time job, that's your problem. No one promised you anything about Unturned's reliability, performance etc.

You should have added my full comment, you are cherrypicking my words, I will add the full comment below, it became my full-time job because of my passion for Unturned and gaming. I was merely clarifying, I'm not here to cause negativity, I'm here because of how frustrated I am with this crappy game. That being said, working on Unturned full-time was a much better experience than slaving away as a warehouse worker. Full comment:

>"I got a bit off base here, but this line of thinking stems from two full years, of hosting Unturned servers as my full time job. At one point I would dream about Unturned, it was the sole focus of my life. Please don't think I'm here just to be negative, I am completely at my wits' end in terms of tolerating this broken-ass shit any longer."

>He gave you this game for free, he gave you and your friends 2 years of enjoyment and you still think he owes you something? He has set up this GitHub issue tracker to take feedback and he responds to every single of them (just verrryy slowly).

It's marketing, dude. We both know damn well the f2p game strategy works quite well, look at a game like Hearthstone for example. Free to play, but generates millions a year in revenue, can we stop pretending that Nelson is making Unturned for the good of us? It's a business bro. Do you make plugins for the "greater good?". LOL. Just because he provided a game for free gives him the right to ignore issues within the game? I think not.

>Like said, player count issue is intended behaviour, the plugins do not fix anything here. They just raise it above the recommended limits. Beside that, this game has an art style. Just because someone uploaded a Ferrari model which does not fit the art style at all, it doesn't mean its better.

Once again, if 24 players is Intended, then we have bigger issues to worry about rather than raising the player count. If 24 players is the INTENDED amount by Nelson, it just further evidences everything I said....? Unless I'm confused...

>Unturned had very poor performance in the past and a lot of memory consumption. This is related to bugs and memory leaks in the Unity version used. However the latest updates fixed this issue. A friend of mine can host 8 full 24 slot servers on 2 core + 4 GB RAM.

The fact you say 2 core, and 4gb ram, just shows how little you know about hosting. What clock speed on CPU? Intel CPU? Architechture? Sandy Bridge, etc? Operating System? What you are saying is irrelevant without knowing all of these variables. I'm glad your friend somehow escapes the performance issues, but for the rest of the 99.9% of community managers, we deal with a laggy, barely playable game.

>These files were meant to be there since it was a new feature, the Community Module. However the module was not registered to Battleye yet so it caused the kickings. As a hotfix Nelson removed it, but he will readd it again.

Let me make this clear. The files added broke the game. If they were intended to be added, was it also intended to break the game? You can easily see where the disconnect in logic happens here.

>Yes Unturneds updates often break stuff, and sometimes Nelson does not fix them fast enough (linux bug for example; which is still around). I agree on Nelson not being really "fast" on fixing stuff or responding to feedback fast.

Well, I'm glad we can agree that broken stuff doesn't get fixed fast enough, which is why I'm here in the first place.


>However people forget that Nelson is a single person, but they expect him to behave like a whole company. He is the community manager, the developer, the art artist, the 2d and 3d artist, etc...

Right, that is what the millions of dollars in revenue is for, to offset your stress and workload. This is basic stuff for anybody that isn't right out of high school.

In conclusion, It's very obvious for anybody on the outside looking in, there is collusion involved. You can't even speak objectively without somebody on Nelson's team attacking you. This is really sad what we have been reduced to.

**Comment by Trojaner25** [comment]
> Glitched bases are hard to fix? That is why a third-party developer made a plugin to stop this happening, 
"Ugly workaround"

>The fact you say 2 core, and 4gb ram, just shows how little you know about hosting. What clock speed on CPU? Intel CPU? Architechture? Sandy Bridge, etc? Operating System? What you are saying is irrelevant without knowing all of these variables. I'm glad your friend somehow escapes the performance issues, but for the rest of the 99.9% of community managers, we deal with a laggy, barely playable game.

I did not include the information you mentioned it isn't necessary on 2 cores. In any way having 8 servers run on only 2 cores with maybe max 4 threads is a good performance. The game has good performance on clients.

>Let me make this clear. The files added broke the game. If they were intended to be added, was it also intended to break the game? You can easily see where the disconnect in logic happens here.

The issue here wasn't that the files got added, it was that they were not registered to BattlEye.

>Once again, if 24 players is Intended, then we have bigger issues to worry about rather than raising the player count. 

It is intended as said by Nelson himself and the Unturned source code proves this. There is no reason to officially support a too high player count.

>You can't even speak objectively without somebody on Nelson's team attacking you. This is really sad what we have been reduced to.

I can say the same about you.

>I was merely clarifying, I'm not here to cause negativity, I'm here because of how frustrated I am with this crappy game

If you don't like it, you can leave :^)

If you really want to make hosting servers your full time job, you should switch to some game which is paid and where some big company is behind it.

**Comment by Alekaei** [comment]
Jim, you are the reason developers dont want to make plugins for unturned.

**Comment by Trojaner25** [comment]
This just proves the "toxic community" part..

**Comment by Alekaei** [comment]
Dear Jim, 
            Its been lonely the past few minutes, I'm growing even more tired and concerned for the safety of you and the people around you. I've been attempting to contain my sadness but its gone too far.

            If you can find it in you to reply please do, I fear I wont see you again.

Yours faithfully,
                           Wizard with the bees knees

**Comment by N1GHTPATROL** [comment]
@Jim-Lahey If you're ever unhappy with the game, you are always able to refund the game for the $0.00 you paid for it.

**Comment by Jim-Lahey** [comment]
Still I have no evidence against any of the claims I've made, just people attacking me personally, it just shows what kind of kids make up the Unturned community. I 100% agree with the toxic part, instead of dealing with the issues themselves, you guys choose to attack me. Professional. Chances are this will be deleted anyways, as everybody gets offended by somebody telling the objective truth.

**Comment by Alekaei** [comment]
Jim, you know who you remind me of, flat earthers, You ignore all the evidence trojaner gave you, and then argue your point. And then complain you dont get any evidence

**Comment by Jonnyfencing1** [comment]
Lol Alekaei

**Comment by Jim-Lahey** [comment]
I would like to point out Trojaner is editing his responses in order to shape how the conversation played out. Sadly, this is what we are reduced to, FUD tactics. Anybody who isn't batting for team SDG can clearly see the millions of dollars in revenue is going to waste on a game, where nobody has any intent to fix the major issues.

**Comment by Trojaner25** [comment]
I always edit my answers afterwards, it has something to do with my perfectionism. You can have a look at my other issues / comments as proof. However it doesn't change the shape of the conversation.

**Comment by Jim-Lahey** [comment]
If you infer objective facts and me outlining them clearly for you, as "aggressive hate speech", it's your problem. You aren't even Nelson and have no bearing over Unturned right? So let Nelson himself respond. Thanks for your input though Trojaner.

**Comment by Trojaner25** [comment]
Just because you think your own opinion is the correct one, it doesnt mean its objective.

**Comment by Jim-Lahey** [comment]
If you have no bearing over Unturned like you say, why do you continue to respond to the official bug tracker? Please stop trying to start a fight, and let's let Nelson respond with his answer to these major issues. If you're not on the Payroll, sorry but your opinion means nothing.

**Comment by Trojaner25** [comment]
You somehow manage to make it even worse.

**Comment by Alekaei** [comment]
just another shitty server manager who thinks hes everything

**Comment by thereis** [comment]
@SDGNelson Resuming... increase the player limit, please!

**Comment by SDGNelson** [comment]
There's a lot of controversial arguments in this thread considering it's about server player limits!

The root issue:

Unturned "supports" 48 players in the same sense that it could "support" 100, but it really does not work very well. The ability to raise it over 24 exists only due to similar requests to be able to see if it works. I think the furthest I'd go is to have a command-line argument or something which enables seeing servers of all sizes.

Other stuff:

Yep: Unturned's multiplayer performance is not very good. Unfortunately 3 is built on a rocky foundation of multiplayer code I wrote when I was 16 that isn't very easily replaceable. As you mention, one of the benefits of transitioning to Unreal is that its netcode is written by experienced network engineers (much smarter than 16/yo me, and today me I expect) and battletested in these popular games like Squad, Ark and PUBG with massive playercounts.

**Comment by thereis** [comment]
I would love to see the servers carrying more than 24 players even using the same engine!

Lets try! @SDGNelson
